src/project/project_manifest_path.rs

```diff
@@ -36,13 +40,54 @@ impl ProjectManifestPath {
     ///
     /// This precedence rule also applies to manifest files themselves.
     /// If there are all `cairo_project.toml`, `Scarb.toml` and `Scarb.lock`
-    /// files in the same directory, the `cairo_project.toml` file will be chosen for each.
-    pub fn discover(path: &Path) -> Option<ProjectManifestPath> {
-        return find_in_parent_dirs(path.to_path_buf(), PROJECT_FILE_NAME)
+    /// files in the same directory, the `cairo_project.toml` file will be chosen for each
+    /// and a notification with a warning will be sent to the client.
+    /// The last step will be skipped for `core` crate to prevent notification spam.
+    pub fn discover(path: &Path, notifier: &Notifier) -> Option<ProjectManifestPath> {
+        let project_config_path = find_in_parent_dirs(path.to_path_buf(), PROJECT_FILE_NAME);
+        let scarb_manifest_path = find_in_parent_dirs(path.to_path_buf(), SCARB_TOML);
+
+        if project_config_path.is_some() && scarb_manifest_path.is_some() {
+            let is_core = match fs::read_to_string(scarb_manifest_path.as_ref().unwrap()) {
+                Ok(content) => {
+                    toml::from_str::<CoreManifest>(&content).unwrap_or_default().package.no_core
```

Comment:
Look for `name == "core"` rather than `no_core = true`. Technically, it's a different thing.